juicefs process will crash when I try to change the  permission of .trash

the panic info  as follows:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1fdb226]

goroutine 1 [running]:
github.com/juicedata/juicefs/pkg/vfs.(*VFS).SetAttr(0xc000398240, {0x2d930b0, 0xc002512000}, 0x7fffffff10000001, 0x1, 0x0, 0x41ff, 0x0, 0x0, 0x0, ...)
    /home/wan/code/juicefs/pkg/vfs/vfs_unix.go:147 +0x1a6
github.com/juicedata/juicefs/pkg/fuse.(*fileSystem).SetAttr(0xc0006154c0, 0x2016e4e, 0xc0009fa1b0, 0xc000478c48)
    /home/wan/code/juicefs/pkg/fuse/fuse.go:102 +0x105
github.com/hanwen/go-fuse/v2/fuse.doSetattr(0xc000478b40, 0xc000478b40)
    /home/wan/go/pkg/mod/github.com/juicedata/go-fuse/v2@v2.1.1-0.20210926080226-cfe1ec802a7f/fuse/opcode.go:195 +0x55
github.com/hanwen/go-fuse/v2/fuse.(*Server).handleRequest(0xc0000de2c0, 0xc000478b40)
    /home/wan/go/pkg/mod/github.com/juicedata/go-fuse/v2@v2.1.1-0.20210926080226-cfe1ec802a7f/fuse/server.go:483 +0x1f3
github.com/hanwen/go-fuse/v2/fuse.(*Server).loop(0xc0000de2c0, 0x0)
    /home/wan/go/pkg/mod/github.com/juicedata/go-fuse/v2@v2.1.1-0.20210926080226-cfe1ec802a7f/fuse/server.go:456 +0x110
github.com/hanwen/go-fuse/v2/fuse.(*Server).Serve(0xc0000de2c0)
    /home/wan/go/pkg/mod/github.com/juicedata/go-fuse/v2@v2.1.1-0.20210926080226-cfe1ec802a7f/fuse/server.go:378 +0x35
github.com/juicedata/juicefs/pkg/fuse.Serve(0xc000398240, {0x0, 0x0}, 0x0)
    /home/wan/code/juicefs/pkg/fuse/fuse.go:460 +0x389
main.mount_main(0xc000398240, 0xc000561890)
    /home/wan/code/juicefs/cmd/mount_unix.go:180 +0x23e
main.mount(0xc0006dcfc0)
    /home/wan/code/juicefs/cmd/mount.go:297 +0x1565
github.com/urfave/cli/v2.(*Command).Run(0xc0007d2360, 0xc0006dc600)
    /home/wan/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/command.go:163 +0x64a
github.com/urfave/cli/v2.(*App).RunContext(0xc00043a4e0, {0x2d4e588, 0xc000136000}, {0xc0000c0200, 0x5, 0x8})
    /home/wan/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:313 +0x81e
github.com/urfave/cli/v2.(*App).Run(...)
    /home/wan/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:224
main.Main({0xc000130000, 0x5, 0x5})
    /home/wan/code/juicefs/cmd/main.go:107 +0x2317
main.main()
    /home/wan/code/juicefs/cmd/main.go:112 +0x2e


wan@wan-VirtualBox:~/code/juicefs$ juicefs -version
juicefs version 1.0-dev (2022-01-25 4b1a394a)

--- a/pkg/vfs/vfs.go
+++ b/pkg/vfs/vfs.go
@@ -385,8 +385,10 @@ func (v *VFS) Open(ctx Context, ino Ino, flags uint32) (entry *meta.Entry, fh ui
 			h.data, _ = json.MarshalIndent(v.Conf, "", " ")
 		}
 		n := getInternalNode(ino)
-		entry = &meta.Entry{Inode: ino, Attr: n.attr}
-		return
+		if n != nil {
+			entry = &meta.Entry{Inode: ino, Attr: n.attr}
+			return
+		}
 	}
 	defer func() {
 		if entry != nil {

--- a/pkg/vfs/vfs_unix.go
+++ b/pkg/vfs/vfs_unix.go
@@ -148,8 +148,10 @@ func (v *VFS) SetAttr(ctx Context, ino Ino, set int, opened uint8, mode, uid, gi
 		n := getInternalNode(ino)
 		if n != nil {
 			entry = &meta.Entry{Inode: ino, Attr: n.attr}
-			return
+		} else {
+			err = syscall.EPERM
 		}
+		return
 	}
 	err = syscall.EINVAL
 	var attr = &Attr{}

--- a/pkg/vfs/vfs_unix.go
+++ b/pkg/vfs/vfs_unix.go
@@ -86,8 +86,10 @@ func (v *VFS) Access(ctx Context, ino Ino, mask int) (err syscall.Errno) {
 	}
 	if IsSpecialNode(ino) {
 		node := getInternalNode(ino)
-		err = accessTest(node.attr, mmask, ctx.Uid(), ctx.Gid())
-		return
+		if node != nil {
+			err = accessTest(node.attr, mmask, ctx.Uid(), ctx.Gid())
+			return
+		}
 	}
 
 	err = v.Meta.Access(ctx, ino, uint8(mmask), nil)
@@ -144,8 +146,10 @@ func (v *VFS) SetAttr(ctx Context, ino Ino, set int, opened uint8, mode, uid, gi
 	}()
 	if IsSpecialNode(ino) {
 		n := getInternalNode(ino)
-		entry = &meta.Entry{Inode: ino, Attr: n.attr}
-		return
+		if n != nil {
+			entry = &meta.Entry{Inode: ino, Attr: n.attr}
+			return
+		}
 	}
 	err = syscall.EINVAL
 	var attr = &Attr{}

